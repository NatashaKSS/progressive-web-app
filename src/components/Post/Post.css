.title {
  font-size: 1.8rem;
}

.content {
  font-size: 1.5rem;
  line-height: 1.5;
}

.link {
  color: #2331de;
}

.link:focus,
.link:hover {
  color: #7e88fc;
  cursor: pointer;
}
