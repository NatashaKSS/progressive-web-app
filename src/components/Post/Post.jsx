import React from 'react';

import PostsAPI from '../../api/PostsAPI.js';

import styles from './Post.css';

const Post = (props) => {
  const id = getPostIDFromProps(props);
  const post = id ? PostsAPI.getPostWithID(id) : null;

  if (!post) {
    return <div>Sorry, the post with this ID was not found!</div>
  } else {
    return (
      <div>
        <h2 className={styles.title}>{post.title}</h2>
        <p className={styles.content}>{post.content}</p>
      </div>
    );
  }
}

/**
 * Obtains the post id from props. Checks if props and nested objects are
 * valid or not, otherwise, an id with a null value will be returned.
 *
 * @param  {Object} props React props
 * @return {String}       String representation of the ID of a post
 */
const getPostIDFromProps = (props) => {
  let id = null;
  if (props.match) {
    if (props.match.params) {
      id = props.match.params.id;
    }
  }
  return id;
}

export default Post;
