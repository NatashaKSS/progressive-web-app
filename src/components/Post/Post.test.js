import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Post from './Post';

describe('<Post />', () => {
  test('<Post /> should render as expected', () => {
    const component = shallow(<Post match={{ params: { id: "010" }}} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('<Post /> should render ID not found as there is no post with this id', () => {
    const component = shallow(<Post match={{ params: { id: "does_not_exist_001" }}} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('<Post /> should render ID not found if no post retrieved', () => {
    const component = shallow(<Post match={{ params: { id: null }}} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  })
});
