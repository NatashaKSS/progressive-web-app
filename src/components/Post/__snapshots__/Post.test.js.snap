// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> <Post /> should render ID not found as there is no post with this id 1`] = `
<div>
  Sorry, the post with this ID was not found!
</div>
`;

exports[`<Post /> <Post /> should render ID not found if no post retrieved 1`] = `
<div>
  Sorry, the post with this ID was not found!
</div>
`;

exports[`<Post /> <Post /> should render as expected 1`] = `
<div>
  <h2
    className="title"
  >
    Apples
  </h2>
  <p
    className="content"
  >
    High in fiber, vitamin C and lots of antioxidants.
  </p>
</div>
`;
